pkg/codegen/internal/test/testdata/external-resource-schema/python/pulumi_example/arg_function.py

```diff
@@ -0,0 +1,54 @@
+# coding=utf-8
+# *** WARNING: this file was generated by test. ***
+# *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import warnings
+import pulumi
+import pulumi.runtime
+from typing import Any, Mapping, Optional, Sequence, Union
+from . import _utilities, _tables
+from pulumi_random import RandomPet
+
+__all__ = [
+    'ArgFunctionResult',
+    'AwaitableArgFunctionResult',
+    'arg_function',
+]
+
+@pulumi.output_type
+class ArgFunctionResult:
+    def __init__(__self__, age=None):
+        if age and not isinstance(age, int):
+            raise TypeError("Expected argument 'age' to be a int")
+        pulumi.set(__self__, "age", age)
+
+    @property
+    @pulumi.getter
+    def age(self) -> Optional[int]:
+        return pulumi.get(self, "age")
+
+
+class AwaitableArgFunctionResult(ArgFunctionResult):
+    # pylint: disable=using-constant-test
+    def __await__(self):
+        if False:
+            yield self
+        return ArgFunctionResult(
+            age=self.age)
+
+
+def arg_function(name: Optional['RandomPet'] = None,
```

Comment:
This seems fine for now. I suspect if we want to actually support this (at least for EKS), as a subsequent change, we'll need to change it so function args that are resources are accepted as Input[RandomPet] and then inside the function do an apply against any input args, and return an output. Per https://github.com/pulumi/pulumi/issues/5385#issuecomment-721325006. But we can cross that bridge when we need to subsequently.